### ISSUE TITLE
any documentation?

### ISSUE THREAD

**Issue by 1010101110** [U-1F31A/BepInEx_Valheim_Full_Updater#2]
so the way this works is it overwrites files in the basic https://valheim.thunderstore.io/package/denikson/BepInExPack_Valheim/ package?

so if i wanted to ensure FULL is installed i would have the bepinex and the full updater package?

and there are no backward compatibility issues?

**Comment by sebastienvercammen** [comment]
@1010101110 That's correct, the Updater overwrites the old pack. Having the updater installed will ensure the Full pack is installed, and the Updater is a plugin so it requires a working BepInEx setup.

There is one issue that exists: If `doorstop_config.ini` is overwritten _after_ installing the Full pack, the Full pack may not be loaded. We've added a feature in the next release of the Updater to automatically fix this (release coming this weekend), so just keep the Updater updated and you'll be good to go.

Some extra notes:

* The Full pack was tested extensively with existing plugins, including big ones like Valheim+ and newer ones that rely on more specialized .NET features
* For backwards compatibility, and compatibility with newer plugins, the DLLs in the pack target .NET 4.5 (released in 2012)
* The Full pack includes all Mono, Mono for Unity, and Unity DLLs, as well as some extras not included in those (e.g. System.Buffers), to be compatible with the features needed by some of the newer plugins — the old pack, for example, won't run with plugins like SebNetCompressionPlugin because it doesn't have the DLLs

There will be no backwards compatibility issues with the old pack.

Documentation for the Updater: https://valheim.thunderstore.io/package/1F31A/BepInEx_Valheim_Full_Updater/
Documentation for the Full pack: 

* https://valheim.thunderstore.io/package/1F31A/BepInEx_Valheim_Full/
* https://github.com/U-1F31A/BepInEx_Valheim_Full

**Comment by 1010101110** [comment]
So in my mod dependency configuration. it would be good to have:
* bepinex
* bepinex full updater

just having bepinex full seems to require the manual install

**Comment by sebastienvercammen** [comment]
@1010101110 That's correct.

r2modman only added support for the Full pack as a BepInEx pack via Thunderstore in v3.1.8, which was 2 days ago. In order not to break people's setups, we're giving everyone at least until after this weekend to run r2mm at least once.

r2mm auto-updates and we don't expect it would cause issues to release today, but it doesn't hurt to err on the side of caution.

After the weekend, everything will be updated: the Updater will have its new features, and the Full pack will be a proper dependency of both the Updater and SebNetCompressionPlugin (and similar plugins).

In your dependencies, you can already refer to the Full pack if you'd like.

**Comment by 1010101110** [comment]
in r2 created a new profile and added bepinex, full updater, and sktoolbox

then click start modded in r2

popup window asking for full update, clicked install

waited for install, click start modded again

![image](https://user-images.githubusercontent.com/4504703/110370176-fac55300-8010-11eb-9d69-2b43a1fcafd3.png)


r2 doesn't start modded, it appears to be starting vanilla. the bepinex console window doesn't start

![image](https://user-images.githubusercontent.com/4504703/110370294-1af51200-8011-11eb-915e-3dcca6561568.png)

**Comment by sebastienvercammen** [comment]
> r2 doesn't start modded, it appears to be starting vanilla. the bepinex console window doesn't start

Try this:

1. Open your r2modman profile folder (you can go there via r2mm's settings)
2. Copy the folder `BepInEx\core_lib` to `core_lib` (so out of `BepInEx`) -- copy, don't move, otherwise the Updater won't find the files
3. Edit `doorstop_config.ini` to say: `dllSearchPathOverride=core_lib` (instead of the old `dllSearchPathOverride=BepInEx\core_lib`

If that fixed it for you, let me know and I'll work to push that update sooner. A handful of people using r2mm on Windows have had this issue, and a permanent fix is included in the update.

Once the update is released, it will update properly even if `dllSearchPathOverride` was changed.

**Comment by 1010101110** [comment]
that fixed it seb

**Comment by 1010101110** [comment]
so are you guys gonna do a release soon?

**Comment by sebastienvercammen** [comment]
Yup, just had an end of our sprint at the day job so it was a busy week wrapping things up. In the meantime, we've also added in a fix for an edge case issue for the Linux startup script.

**Comment by 1010101110** [comment]
i tried the update today but  still seem to have same issue
